tutorials/kotlin-springboot-app-engine-java8.md

```diff
@@ -0,0 +1,227 @@
+
+    > Alternatively, you can [download](https://github.com/GoogleCloudPlatform/kotlin-samples/tree/master/appengine/springboot/) the sample application.
+
+1. Download the generated project and save it to a local folder.
+
+1. Open the resulting project in your favourite IDE or editor and create a new
+source file `MessageController.kt` in the directory
+`src/main/kotlin/com/example/demo` with the following contents:
+    ```kt
+    package com.example.demo
+
+    import org.springframework.web.bind.annotation.RequestMapping
+    import org.springframework.web.bind.annotation.RestController
+
+    data class Message(val text: String, val priority: String)
+
+    @RestController
+    class MessageController {
+        @RequestMapping("/message")
+        fun message(): Message {
+            return Message("Hello from Google Cloud", "High")
+        }
+    }
+    ```
+
+    The package should match that of your group and artifact name.
+
+1.  Edit the file named `DemoApplication.kt` in
+    `src/main/kotlin/com/example/demo` and replace its contents with the
+    following:
+    ```kt
+    package com.example.demo
+
+    import org.springframework.boot.autoconfigure.SpringBootApplication
+    import org.springframework.boot.runApplication
+    import org.springframework.boot.web.servlet.support.SpringBootServletInitializer
+
+    @SpringBootApplication
+    class DemoApplication : SpringBootServletInitializer()
+
+    fun main(args: Array<String>) {
+        runApplication<DemoApplication>(*args)
+    }
+    ```
+
+1. Run the application from the command line using Maven:
+
+        mvn spring-boot:run
+
+1. Open the browser and make sure you get a valid JSON response when accessing http://localhost:8080/message. The result should be:
+    ```json
+    {
+      "text": "Hello from Google Cloud",
+      "priority": "High"
+    }
+    ```
+
+## Deploy your application
+
+To deploy your application, you will use an App Engine plugin for Maven which simplifies some of the process. The plugin
+is also [available for Gradle](https://cloud.google.com/appengine/docs/standard/java/tools/gradle).
+
+1.  Create a file called `appengine-web.xml` in a new folder `src/main/webapp/WEB-INF` with the following contents:
+    ```xml
+    <?xml version="1.0" encoding="utf-8"?>
+    <appengine-web-app xmlns="http://appengine.google.com/ns/1.0">
+        <threadsafe>true</threadsafe>
+        <runtime>java8</runtime>
+    </appengine-web-app>
+    ```
+
+1.  Add the following entry to the `plugins` section of your `pom.xml` file to
+    configure the [Google Cloud plugin](https://cloud.google.com/appengine/docs/standard/java/tools/maven):
+    ```xml
+    <plugin>
+        <groupId>com.google.cloud.tools</groupId>
+        <artifactId>appengine-maven-plugin</artifactId>
+        <version>1.3.1</version>
+    </plugin>
+    ```
+
+1.  Run the following command to deploy your app:
+
+        mvn appengine:deploy
```

Comment:
Again, sorry for my ignorance, but do you need an application name or anything for this command, or has it been set up sufficiently in previous steps?

Reply:
It is not necessary to supply an application name. The project ID is determined by your credentials and everything else is defined in`application-web.xml`